### PR TITLE
Closes #3746 remove ForwardRef

### LINKED ISSUE #3746: remove ForwardRef
Remove `ForwardRef` from the code.

### PR DESCRIPTION
**Pull request by ajpotts**
Closes #3746 remove ForwardRef

### CODE REVIEW

File: arkouda/pdarraysetops.py [hunk 1]
```diff
@@ -90,9 +95,9 @@ def _in1d_single(
         # this causes an issue with mypy for unknown reasons.
         if pda1.size == 0:
             return zeros(0, dtype=akbool)
-    if isinstance(pda2, pdarray) or isinstance(pda2, Strings) or isinstance(pda2, Categorical_):
-        if pda2.size == 0:
-            return zeros(pda1.size, dtype=akbool)
+        if isinstance(pda2, pdarray) or isinstance(pda2, Strings) or isinstance(pda2, Categorical_):
+            if pda2.size == 0:
+                return zeros(pda1.size, dtype=akbool)
```
Comment (stress-tess) [inline arkouda/pdarraysetops.py:100]: was this meant to be indented?

**Review by stress-tess — COMMENTED** [review]
This is awesome!!! I've tried a couple times to get rid of those forward refs and never made it happen! `TypeVar` seems pretty cool! I read up on it a bit here:
https://dev.to/decorator_factory/typevars-explained-hmo

The one thing I saw was we indented an if block and I want to make sure it was on purpose

File: arkouda/pdarraysetops.py (hunk 1 above)
Comment (ajpotts) [reply arkouda/pdarraysetops.py:100]: Yes, if you look at the logic it only makes sense if both `pda1` and `pda2` are `(pdarray, Strings, Categorical)` (not a Sequence), which mypy will complain about.  Indenting was the easy way of achieving this.

**Review by stress-tess — APPROVED** [review]
Looks good!

File: arkouda/pdarraysetops.py (hunk 1 above)
Comment (stress-tess) [reply arkouda/pdarraysetops.py]: makes sense! I like the change!

**Pull request merged**